Mathlib/FieldTheory/IsPerfectClosure.lean

```diff
@@ -0,0 +1,421 @@
+      (h _ (pow_one p ▸ natSepDegree_X_pow_char_sub_C p 1 x))
+      ((degree_X_pow_sub_C (expChar_pos F p) x).symm ▸ Nat.cast_pos.2 (expChar_pos F p)).ne'
+    exact ⟨y, by rwa [← eval, eval_sub, eval_pow, eval_X, eval_C, sub_eq_zero] at hy⟩
+  exact PerfectRing.toPerfectField F p
+
+-- TODO: move to suitable location
+theorem iterate_frobeniusEquiv_symm_apply_iterate_frobenius (R : Type*) (p : ℕ)
+    [CommSemiring R] [ExpChar R p] [PerfectRing R p] (x : R) (n : ℕ) :
+    (frobeniusEquiv R p).symm^[n] ((frobenius R p)^[n] x) = x := by
+  induction n with
+  | zero => rfl
+  | succ n ih => rwa [Function.iterate_succ_apply, Function.iterate_succ_apply',
+    frobeniusEquiv_symm_apply_frobenius]
+
+-- TODO: move to suitable location
+theorem iterate_frobenius_apply_iterate_frobeniusEquiv_symm (R : Type*) (p : ℕ)
+    [CommSemiring R] [ExpChar R p] [PerfectRing R p] (x : R) (n : ℕ) :
+    (frobenius R p)^[n] ((frobeniusEquiv R p).symm^[n] x) = x := by
+  induction n with
+  | zero => rfl
+  | succ n ih => rwa [Function.iterate_succ_apply, Function.iterate_succ_apply',
+    frobenius_apply_frobeniusEquiv_symm]
+
+-- TODO: move to suitable location
+theorem iterate_frobeniusEquiv_symm_mul (R : Type*) (p : ℕ)
+    [CommSemiring R] [ExpChar R p] [PerfectRing R p] (x y : R) (n : ℕ) :
+    (frobeniusEquiv R p).symm^[n] (x * y) =
+    (frobeniusEquiv R p).symm^[n] x * (frobeniusEquiv R p).symm^[n] y := by
+  induction n with
+  | zero => rfl
+  | succ n ih => rw [Function.iterate_succ_apply', ih, map_mul,
+    Function.iterate_succ_apply', Function.iterate_succ_apply']
+
+-- TODO: move to suitable location
+theorem iterate_frobeniusEquiv_symm_add (R : Type*) (p : ℕ)
+    [CommSemiring R] [ExpChar R p] [PerfectRing R p] (x y : R) (n : ℕ) :
+    (frobeniusEquiv R p).symm^[n] (x + y) =
+    (frobeniusEquiv R p).symm^[n] x + (frobeniusEquiv R p).symm^[n] y := by
+  induction n with
+  | zero => rfl
+  | succ n ih => rw [Function.iterate_succ_apply', ih, map_add,
+    Function.iterate_succ_apply', Function.iterate_succ_apply']
+
+-- TODO: move to suitable location
+theorem iterate_frobeniusEquiv_symm_pow (R : Type*) (p : ℕ)
+    [CommSemiring R] [ExpChar R p] [PerfectRing R p] (x : R) (n m : ℕ) :
+    (frobeniusEquiv R p).symm^[n] (x ^ m) =
+    (frobeniusEquiv R p).symm^[n] x ^ m := by
+  induction n with
+  | zero => rfl
+  | succ n ih => rw [Function.iterate_succ_apply', ih, map_pow, Function.iterate_succ_apply']
+
+-- TODO: move to suitable location
+theorem iterate_frobeniusEquiv_symm_zero (R : Type*) (p : ℕ)
+    [CommSemiring R] [ExpChar R p] [PerfectRing R p] (n : ℕ) :
+    (frobeniusEquiv R p).symm^[n] 0 = 0 := by
+  induction n with
+  | zero => rfl
+  | succ n ih => rw [Function.iterate_succ_apply', ih, map_zero]
+
+-- TODO: move to suitable location
+theorem iterate_frobeniusEquiv_symm_one (R : Type*) (p : ℕ)
+    [CommSemiring R] [ExpChar R p] [PerfectRing R p] (n : ℕ) :
+    (frobeniusEquiv R p).symm^[n] 1 = 1 := by
```

Comment:
I think most of the above lemmas are needed if you use [iterateFrobeniusEquiv](https://leanprover-community.github.io/mathlib4_docs/Mathlib/FieldTheory/Perfect.html#iterateFrobeniusEquiv) I just introduced in #10016, which is already a RingHom. I added [two more lemmas](https://github.com/leanprover-community/mathlib4/pull/9311/files#diff-e161d5e9d8af730e6a06bd5fb3f86e514a748447e287d2c2c9e9a2e95ee0769eR90-R96) in #9311.

Reply:
Would you not define it as `(iterateFrobeniusEquiv R p n).symm`?